### PR TITLE
FISH-205 Implement Dynamic Micro Logging Properties

### PR DESCRIPTION
**Pull request by MattGill98**
# Description

When running Payara Micro with `--enableDynamicLogging` the logging properties file will be copied into the exploded directory and read again.

# Testing

To reproduce, use the following log properties file:

~~~
#Payara Micro Logging Properties File

## Handlers
handlers=java.util.logging.ConsoleHandler
java.util.logging.ConsoleHandler.formatter=com.sun.enterprise.server.logging.ODLLogFormatter
com.sun.enterprise.server.logging.ODLLogFormatter.ansiColor=true
java.util.logging.ConsoleHandler.level=FINE

## Global Level
.level=INFO

#All log level details
PayaraMicro.level=INFO
~~~

Run Payara Micro with --logproperties and notice that changing the file doesn't print anything.
Run Payara Micro also with --enableDynamicLogging and change the global log level to FINE. Notice that an INFO log appears saying log levels have been updated. Change the global log level back to INFO, and notice a new FINE level message has been logged.

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins test please

**Comment by MattGill98** [comment]
jenkins test please

File: nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraFileWatcher.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                watcher.close();
+                running = false;
+                LOGGER.info("Terminated the file watcher service");
+            } catch (IOException ex) {
+                LOGGER.log(WARNING, "Failed to terminate the watch service", ex);
+            }
+        }
+    }
+
+    // Lifecycle methods
```
Comment (Pandrex247) [inline nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraFileWatcher.java:196]: Could these not be at the top?
This is edging dangerously close to the mindset of class variables being declared at the bottom, and that's an abyss that's swallowed many a good developer.

Seriously though they're at the top in most other classes.

Comment (MattGill98) [reply nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraFileWatcher.java:196]: I haven't seen it explicitly declared in any of our code styling comments so I assumed it was open to interpretation! To me, these methods belong at the back, but I'm happy to move them to the top

Comment (Pandrex247) [reply nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraFileWatcher.java:196]: You're right they don't seem to be mentioned in the Sun code conventions (or if they are they're not where I expect them to be).

I'd still prefer it if _PostConstruct_ at least was at the top though - in most occurrences it seems to be near the top (_PreDestroy_ is all over the place lol).
I tend to count lifecycle methods in a similar vein to constructors, if anything a constructor is also a lifecycle method.
_PostConstruct_ I'd always expect to find at the top - when reading the code (and debugging) it means you don't start at the top, jump to the bottom, then go back to the top again.

**Comment by MattGill98** [comment]
I'll make all of the code style changes once I've had a functional review!

**Comment by MattGill98** [comment]
jenkins test please

**Pull request merged**